src/socket.rs

```diff
@@ -1391,6 +1397,10 @@ mod test {
         ($e:expr) => (match $e { Ok(e) => e, Err(e) => panic!("{:?}", e) })
     }
 
+    macro_rules! mutetry {
```

Comment:
potentially dangerous, yes prints a log message but really we should handle the error